spec/requests/flavors_spec.rb

```diff
@@ -0,0 +1,163 @@
+RSpec.describe 'Flavors API' do
+  let(:ems) { FactoryGirl.create(:ems_cloud) }
+  let(:flavor) { FactoryGirl.create(:flavor_openstack, :ext_management_system => ems) }
+  let(:flavor_2) { FactoryGirl.create(:flavor_openstack, :ext_management_system => ems) }
+
+  describe 'GET/api/flavors' do
+    it 'lists all the flavor bases with an appropriate role' do
+      flavor = FactoryGirl.create(:flavor_openstack)
+
+      api_basic_authorize collection_action_identifier(:flavors, :read, :get)
+
+      get(api_flavors_url)
+
+      expected = {
+        'count'     => 1,
+        'subcount'  => 1,
+        'name'      => 'flavors',
+        'resources' => [hash_including('href' => a_string_matching(api_flavors_url(nil, flavor)))]
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'forbids access to flavor bases without an appropriate role' do
+      api_basic_authorize
+
+      get(api_flavors_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe 'GET /api/flavors/:id' do
+    it 'will show a flavor base' do
+      api_basic_authorize action_identifier(:flavors, :read, :resource_actions, :get)
+
+      get(api_flavor_url(nil, flavor))
+
+      expected = {
+        'href' => a_string_matching(api_flavors_url(nil, flavor))
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'forbids access to a flavor base' do
+      api_basic_authorize
+
+      get(api_flavors_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe 'DELETE /api/flavors/:id' do
+    it 'will delete a flavor' do
+      api_basic_authorize action_identifier(:flavors, :delete, :resource_actions, :delete)
+
+      delete api_flavor_url(nil, flavor)
+
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it 'will not delete a flavor without an appropriate role' do
+      api_basic_authorize
+
+      delete api_flavor_url(nil, flavor)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe 'POST /api/flavors' do
+    let(:create_params) do
+      {
+        :action      => 'create',
+        :description => "Description",
+        :name        => "A Flavor",
+        :related     => {},
+        :ems         => ems,
```

Comment:
a little confused how this is working. should this be `ems => { 'href' => api_provider_url(nil, ems) }`? One last minor request as well, could you add a test for the validation of the ems provider? Thanks! 👍